### ISSUE TITLE
holonomic tests fail in Python 3.6

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#11607]
I haven't checked why. They pass in Python 3.5:

```
___________________________________________________ sympy/holonomic/tests/test_holonomic.py:test_to_hyper ____________________________________________________
  File "/users/aaronmeurer/documents/python/sympy/sympy/sympy/holonomic/tests/test_holonomic.py", line 531, in test_to_hyper
    assert p == q
AssertionError
______________________________________________________________________________________________________________________________________________________________
_____________________________________ sympy/holonomic/tests/test_holonomic.py:test_extended_domain_in_expr_to_holonomic ______________________________________
  File "/users/aaronmeurer/documents/python/sympy/sympy/sympy/holonomic/tests/test_holonomic.py", line 692, in test_extended_domain_in_expr_to_holonomic
    assert p - x - 1.2*cos(1.0*x) == 0
AssertionError
```

@shubhamtibra

**Comment by shubhamtibra** [comment]
Thanks for testing @asmeurer . My college exams are currently going on so I might not be able to look it up before one and a half week. What's our deadline for providing Python 3.6 support?

**Comment by asmeurer** [comment]
The final release is supposed to come out in [december](https://www.python.org/dev/peps/pep-0494/). It would be nice to have this fixed by then. Sooner if we decide to do a SymPy release before then, as we'd want it to support Python 3.6. I've tagged this with the 1.1 milestone so that we don't forget to get it fixed for the next release.

**Comment by jksuom** [comment]
The two for loops for [a-parameters](https://github.com/sympy/sympy/blob/master/sympy/holonomic/holonomic.py#L2016) and [b-parameters](https://github.com/sympy/sympy/blob/master/sympy/holonomic/holonomic.py#L2019) of a hypergeomeric series determine the order in the output. Since `arg1` and `arg2` are dictionaries, the order should be made deterministic by something like `for k in ordered(arg1.keys())`.

**Comment by shubhamtibra** [comment]
Thanks @jksuom . I have opened up a pull request.

**Comment by jksuom** [comment]
Fixed by #11652
